### PR TITLE
✨ Source MongoDB Internal POC: Add CDC-focused Acceptance Tests

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Add acceptance tests to verify CDC behavior, specifically with regards to offset tracking

## How
* Add CDC-focused acceptance tests inspired by `CdcPostgresSourceTest`

## Recommended reading order
1. `MongoDbSourceAcceptanceTest.java`

**N.B.** the `testSyncEmptyCollection` test will fail until the identified issue with syncing an empty collection is fixed.

### CODE REVIEW

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java [hunk 1]
```diff
@@ -162,6 +204,165 @@ public void testIncrementalReadSelectedColumns() throws Exception {
...
+    final ConfiguredAirbyteCatalog configuredCatalog = getConfiguredCatalog();
+    final List<ConfiguredAirbyteStream> streams = configuredCatalog.getStreams();
+    final AirbyteStream otherAirbyteStream = MongoCatalogHelper.buildAirbyteStream(OTHER_COLLECTION_NAME, DATABASE_NAME,
+        List.of(Field.of(NAME_FIELD, JsonSchemaType.STRING), Field.of(INT_TEST_FIELD, JsonSchemaType.NUMBER)));
+    streams.add(convertToConfiguredAirbyteStream(otherAirbyteStream, SyncMode.INCREMENTAL));
+    configuredCatalog.withStreams(streams);
+
+    // Start a sync with two streams
+    final List<AirbyteMessage> messages = runRead(configuredCatalog);
+    final List<AirbyteRecordMessage> recordMessages = filterRecords(messages);
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java:241]: Can we assert the contents of the record message as well. Also assert that the record messages contain the metadata (like the cursor field value) that should be added.

**Review by subodh1810 — COMMENTED** [review]
This looks good. Do we have tests for the following scenario? 
1. INSERT, UPDATE, DELETE events produce right records in incremental (second) sync
2. A new table is added to a sync. (first sync runs with 1 table, second sync runs with 2 table but with state emitted from first sync)
3. no data for incremental sync i.e. first sync runs and outputs a state, second sync runs with that state but there has been no changes in the source to capture so the second sync should return 0 records.

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java [hunk 2]
```diff
@@ -177,4 +378,38 @@ private void verifyFieldNotExist(final List<AirbyteRecordMessage> records, final
...
+  private Document createDocument(final Integer i) {
+    return new Document(NAME_FIELD, "value" + i).append(INT_TEST_FIELD, i);
+  }
+
+  private void validateStateMessages(final List<AirbyteStateMessage> stateMessages) {
+    stateMessages.forEach(stateMessage -> {
+      assertEquals(AirbyteStateMessage.AirbyteStateType.GLOBAL, stateMessage.getType());
+      final AirbyteGlobalState global = stateMessage.getGlobal();
+      assertNotNull(global.getSharedState());
+      assertFalse(global.getSharedState().isEmpty());
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java:403]: can we validate that the shared state should only contain the specific streams i.e. the first state should not contain the second stream and only contain the first stream, etc.

**Review by subodh1810 — APPROVED** [review]
Based on our review call discussion we need to :
1. Assert cdc metadata for records that are pulled in during the initial sync as well
2. Assert the checkpoint state message emitted during the initial sync of a stream (with a catalog containing multiple streams) should only contain the stream that is being synced and the ones that have finished. Not the ones that are yet to be synced.
3. Regarding, DebeziumEventUtils I think there is scope of getting rid of an extra serialisation (although not sure). Since serialisation call will be done for each record and we have multiple instances of serialisation/de-serialisation, it can have a performance impact so if possible let's try to minimise it. 
4. Discover call needs to fixed (based on our slack conversation around the `_id` being the primary key not the cursor field and `_ab_cdc_cursor` being marked as the cursor field)
5. With the changes around the cdc module let's run the tests for Postgres, Mysql and MSSQL as well just to make sure we are not causing any regression. 

Once the above points are taken care of then I think the PR is ready. Amazing effort!!

**Comment by jdpgrailsdev** [comment]
> Based on our review call discussion we need to :
> 
> 1. Assert cdc metadata for records that are pulled in during the initial sync as well
> 2. Assert the checkpoint state message emitted during the initial sync of a stream (with a catalog containing multiple streams) should only contain the stream that is being synced and the ones that have finished. Not the ones that are yet to be synced.
> 3. Regarding, DebeziumEventUtils I think there is scope of getting rid of an extra serialisation (although not sure). Since serialisation call will be done for each record and we have multiple instances of serialisation/de-serialisation, it can have a performance impact so if possible let's try to minimise it.
> 4. Discover call needs to fixed (based on our slack conversation around the `_id` being the primary key not the cursor field and `_ab_cdc_cursor` being marked as the cursor field)
> 5. With the changes around the cdc module let's run the tests for Postgres, Mysql and MSSQL as well just to make sure we are not causing any regression.
> 
> Once the above points are taken care of then I think the PR is ready. Amazing effort!!

Items 3 and 4 have been fixed on this branch.  1,2 an 5 will be done against the cdc-integration branch (the target of this PR) after merge.

**Pull request merged**